tests/rustdoc-js/self-is-not-generic.js

```diff
@@ -0,0 +1,22 @@
+// exact-check
+
+const EXPECTED = [
+    {
+        'query': 'A -> A',
+        'others': [
+            { 'path': 'self_is_not_generic::Thing', 'name': 'from' }
```

Comment:
I agree we should filter out blanket impls, but it's looking to be a more involved change than I expected at first. Would you be ok to land this since it's strictly an improvement, and then I can fix blanket impls in a followup change? I think I'll have to refactor the `Cache` code since it's very buggy.